lib/mapObjects/CQuest.h

```diff
@@ -17,49 +17,27 @@ VCMI_LIB_NAMESPACE_BEGIN
 
 class CGCreature;
 
-class DLL_LINKAGE CQuest final
+class DLL_LINKAGE CQuest: public Rewardable::Limiter
 {
-	mutable std::unordered_map<ArtifactID, unsigned, ArtifactID::hash> artifactsRequirements; // artifact ID -> required count
-
 public:
-	enum Emission {
-		MISSION_NONE = 0,
-		MISSION_LEVEL = 1,
-		MISSION_PRIMARY_STAT = 2,
-		MISSION_KILL_HERO = 3,
-		MISSION_KILL_CREATURE = 4,
-		MISSION_ART = 5,
-		MISSION_ARMY = 6,
-		MISSION_RESOURCES = 7,
-		MISSION_HERO = 8,
-		MISSION_PLAYER = 9,
-		MISSION_HOTA_MULTI = 10,
-		// end of H3 missions
-		MISSION_KEYMASTER = 100,
-		MISSION_HOTA_HERO_CLASS = 101,
-		MISSION_HOTA_REACH_DATE = 102
-	};
 
-	enum Eprogress {
+	enum EProgress {
 		NOT_ACTIVE,
 		IN_PROGRESS,
 		COMPLETE
 	};
 
-	static const std::string  & missionName(Emission mission);
-	static const std::string  & missionState(int index);
+	static const std::string & missionName(int index);
+	static const std::string & missionState(int index);
+	
+	std::string questName;
 
 	si32 qid; //unique quest id for serialization / identification
 
-	Emission missionType;
-	Eprogress progress;
+	EProgress progress;
 	si32 lastDay; //after this day (first day is 0) mission cannot be completed; if -1 - no limit
-
-	ui32 m13489val;
-	std::vector<ui32> m2stats;
-	std::vector<ArtifactID> m5arts; // artifact IDs. Add IDs through addArtifactID(), not directly to the field.
-	std::vector<CStackBasicDescriptor> m6creatures; //pair[cre id, cre count], CreatureSet info irrelevant
-	TResources m7resources;
+	int killTarget;
```

Comment:
I think this is ID of map object instance?
```suggestion
	ObjectInstanceID killTarget;
```